### ISSUE TITLE
mpz += 0 == random value

### ISSUE THREAD

**Issue by jamesjer** [aleaxit/gmpy#200]
Due to an upgrade to sagemath 8.2 in Fedora, I had to update gmpy2 to 2.1.0alpha2 recently.  Now sympy is failing tests.  The root cause appears to be this:

```
>>> from gmpy2 import mpz
>>> result = mpz(0)
>>> print result
0
>>> result += long(0)
>>> print result
3224158496
```

It looks like this happens because in GMPy_MPZ_IAdd_Slot, in the switch statement starting on line 48, cases 0 and 1 are handled identically.  But that is wrong.  If the size is 0, then ob_digit[0] contains random bytes.  The value 0 needs to be passed in explicitly in that case.

**Comment by skirpichev** [comment]
@casevh, probably this was fixed in gmpy2 2.1.0a3?

**Comment by casevh** [comment]
Yes, this was fixed in gmpy2 2.1.0a3. Closing....
